src/lib/select/select.spec.ts

```diff
@@ -62,3203 +70,3284 @@ describe('MatSelect', () => {
-          .toEqual('steak-0', `Expected control's value to be set to the new option.`);
-    }));
+          expect(options[0].getAttribute('aria-disabled')).toEqual('false');
+          expect(options[1].getAttribute('aria-disabled')).toEqual('false');
+          expect(options[2].getAttribute('aria-disabled')).toEqual('false');
+        }));
+      });
 
-    it('should clear the selection when a nonexistent option value is selected', fakeAsync(() => {
-      fixture.componentInstance.control.setValue('pizza-1');
-      fixture.detectChanges();
+      describe('for option groups', () => {
+        let fixture: ComponentFixture<SelectWithGroups>;
+        let trigger: HTMLElement;
+        let groups: NodeListOf<HTMLElement>;
 
-      fixture.componentInstance.control.setValue('gibberish');
-      fixture.detectChanges();
+        beforeEach(fakeAsync(() => {
+          fixture = TestBed.createComponent(SelectWithGroups);
+          fixture.detectChanges();
+          trigger = fixture.debugElement.query(By.css('.mat-select-trigger')).nativeElement;
+          trigger.click();
+          fixture.detectChanges();
+          groups =
+              overlayContainerElement.querySelectorAll('mat-optgroup') as NodeListOf<HTMLElement>;
+        }));
 
-      const value = fixture.debugElement.query(By.css('.mat-select-value'));
-      expect(value.nativeElement.textContent.trim())
-          .toBe('', `Expected trigger to be cleared when option value is not found.`);
-      expect(trigger.textContent)
-          .not.toContain('Pizza', `Expected trigger to be cleared when option value is not found.`);
+        it('should set the appropriate role', fakeAsync(() => {
+          expect(groups[0].getAttribute('role')).toBe('group');
+        }));
 
-      trigger.click();
-      fixture.detectChanges();
-      flush();
+        it('should set the `aria-labelledby` attribute', fakeAsync(() => {
+          let group = groups[0];
+          let label = group.querySelector('label')!;
 
-      const options =
-          overlayContainerElement.querySelectorAll('mat-option') as NodeListOf<HTMLElement>;
-      expect(options[1].classList)
-          .not.toContain('mat-selected', `Expected option with the old value not to be selected.`);
-    }));
+          expect(label.getAttribute('id')).toBeTruthy('Expected label to have an id.');
+          expect(group.getAttribute('aria-labelledby'))
+              .toBe(label.getAttribute('id'), 'Expected `aria-labelledby` to match the label id.');
+        }));
 
+        it('should set the `aria-disabled` attribute if the group is disabled', fakeAsync(() => {
+          expect(groups[1].getAttribute('aria-disabled')).toBe('true');
+        }));
+      });
 
-    it('should clear the selection when the control is reset', fakeAsync(() => {
-      fixture.componentInstance.control.setValue('pizza-1');
-      fixture.detectChanges();
+      describe('aria-owns', () => {
```

Comment:
Is this one supposed to be empty?

Reply:
Fixed- I copied the contents to a different `describe`

Reply:
It seems like the empty `describe` is still there.